### DISCUSSION TITLE
sql functions return null with local supabase

### DISCUSSION THREAD

**Discussion by tedhsiehhydrate** [supabase/supabase#22723] [Questions]
Hi, I'm new to the Supabase. While I'm learning and creating my own sql functions with local Supabase , they all return null (undefined) data without any error. But they are working fine on the Supabse Dashboard.
After googling online, seems like RLS is the issue that most people are facing. However, none of my table has RLS enabled. So I used the most basic sql function I can get

```
create or replace function hello_world() returns text as $$
  select 'Hello world';
$$ language sql;
```

And my client side is still getting null though. 
```
const { data, error } = await supabase
    .rpc('hello_world')

    if (error) {
        console.error('Error calling hello_world:', error);
        return null;
    } else {
        console.log('hello_world:', data);
    }
```
Result
```
hello_world: undefined
```

Note that I'm able to use supabase lib to do other sql operations, it is just that sql functions seem always return null.

**Comment by GaryAustin1** [comment]
I ran your function SQL.
And tested with this:
```js
    const { data, error } = await supabases[2].rpc('hello_world')
    console.log(data,error)
```
 And get back    
![image](https://github.com/supabase/supabase/assets/54564956/16e26726-f49c-4be7-88d9-a468aebe2f76)


Any chance you have changed the function to fix something and something like next.js is caching your original not working query?

**Comment by tedhsiehhydrate** [comment]
Hi @GaryAustin1,

I didn't change the function, I have been testing it for hours, and I'm running it with a plain javascript program though.
This function works fine whenever I directly call it from the Supabse SQL editor.
<img width="1260" alt="Screenshot 2024-04-15 at 8 43 19 AM" src="https://github.com/supabase/supabase/assets/166192547/bfdf6fe9-7aac-4f9a-8fd3-0f282d3e46b7">

I'm able to interact with the database thru supabase-js from my javacript code
```
let { data: users, error } = await supabase
    .from("users")
    .select("*")
    .eq("role", 2);

  if (error) {
    console.error("Error fetching users:", error);
    return null;
  }
```
It returns the correct result ( These are random users I created for testing)
```
[
  {
    first_name: 'Lucy',
    last_name: 'Wilson',
    dob: '1988-11-11',
    role: 2,
    user_pic_url: 'https://example.com/photos/lucywilson.jpg',
    subscription_type: 0,
    created_at: '2024-04-15T15:12:26.27019',
    updated_at: '2024-04-15T15:12:26.27019',
    deleted_at: null
  },
  {
    first_name: 'Emma',
    last_name: 'Thomas',
    dob: '1993-02-22',
    role: 2,
    user_pic_url: 'https://example.com/photos/emmathomas.jpg',
    subscription_type: 0,
    created_at: '2024-04-15T15:12:26.329272',
    updated_at: '2024-04-15T15:12:26.329272',
    deleted_at: null
  },
```
I also noticed one thing that if I issue the rpc to a non-existed function call, it still returns null instead of an error.
```
const { data, error } = await supabase
    .rpc('hello_world_non_exist')

    if (error) {
        console.error('Error calling hello_world:', error);
        return null;
    } else {
        console.log('hello_world:', data);
    }
```
```
hello_world: undefined
```

**Reply by GaryAustin1** [reply]
Something is odd then.... LOL sorry
![image](https://github.com/supabase/supabase/assets/54564956/7c5833f8-a11e-46f5-8ee9-f89485ded615)
Is what you get with bad function name...
My test platform is just JS.

Can you see the network tab and the request go out and come back?  

I am not local hosting Supabase though, but still would expect an error.

**Reply by tedhsiehhydrate** [reply]
I couldn't find any rpc request reaches there, but i was able to see the below log in kong logs.
```
2024-04-15 15:38:47 192.168.65.1 - - [15/Apr/2024:22:38:47 +0000] "POST /rest/v1/rpc/hello_world HTTP/1.1" 200 23 "-" "node"
```

<img width="1487" alt="image" src="https://github.com/supabase/supabase/assets/166192547/1f2cc3a4-0da1-417e-b3b0-e4bcc2c08a19">
I couldn't find anything else from other places apart from this.


wondering if there are any settings that I have to set in order to use/enable supabase rpc?

**Reply by GaryAustin1** [reply]
It is just a database operation but does use POST versus select which uses GET.
If insert works, it would work (not counting RLS differences).
There are no settings.

**Reply by tedhsiehhydrate** [reply]
insert works in my code, i do inserts in multiple places in my code and they all work. I can see the data is reflected is the Supabase Dashboard Database page and I can also fetch the data from my code too.

I had excluded RLS factors as i'm only using the basic hello_world function though.

**Reply by GaryAustin1** [reply]
Sorry I've got nothing else.  I don't do local or self-hosted so don't really know what you could do to impact just rpc.  I can't think of anything.

**Reply by tedhsiehhydrate** [reply]
ok, thank you @GaryAustin1 .

btw, may I know which version of supabase-js you are using?

**Reply by GaryAustin1** [reply]
The latest.  2.42.4

**Comment by tedhsiehhydrate** [comment]
Update:

I was able to call the functions from curl, but still can't get any results from supabase.rpc api..
```
➜  ~ curl -X POST 'http://127.0.0.1:54321/rest/v1/rpc/hello_world' \
-H "apikey: ***"
"Hello world"%
➜  ~ curl -X POST 'http://127.0.0.1:54321/rest/v1/rpc/hello_world_non_exist' \
-H "apikey: ***"
{"code":"PGRST202","details":"Searched for the function public.hello_world_non_exist without parameters, but no matches were found in the schema cache.","hint":"Perhaps you meant to call the function public.hello_world","message":"Could not find the function public.hello_world_non_exist without parameters in the schema cache"}%
```

**Reply by GaryAustin1** [reply]
I noticed you closed this.  Did you just give up or find the answer?

**Reply by tedhsiehhydrate** [reply]
It started working after I rebooted my laptop. I didn't figure out what was going on, but it works now!
Thanks @GaryAustin1
